src/Core/View.js

```diff
@@ -317,8 +363,26 @@ View.prototype.addLayer = function addLayer(layer, parentLayer) {
         this.scene.add(layer.object3d);
     }
 
-    this.notifyChange(true);
-    return layer.whenReady;
+    if (this._readyCallback) {
+        this.removeFrameRequester(MAIN_LOOP_EVENTS.AFTER_RENDER, this._readyCallback);
+    }
+
+    return layer.whenReady.then((layer) => {
+        this.notifyChange(false);
+
+        if (!this._readyCallback) {
+            this._readyCallback = () => {
+                if (this.mainLoop.scheduler.commandsWaitingExecutionCount() == 0) {
```

Comment:
it may not be enough. Elements may be waiting to be update in `viewer._changeSources`
it might be necessary to add in test `viewer._changeSources.size == 0` to make sure the queue isn't going to fill up again

Reply:
I fixed this but I used the `RENDERING_SCHEDULED` state of the mainloop instead.